tests/backend/mongodb/test_changefeed.py

```diff
@@ -1 +1,168 @@
+                                     mock.DEFAULT, mock.DEFAULT, False]
+    mock_cursor_next.side_effect = [mock.DEFAULT] + mock_changefeed_data
+    mock_cursor_find_one.return_value = mock_changefeed_data[2]['o']
+
+    outpipe = Pipe()
+    changefeed = get_changefeed(conn, 'backlog', ChangeFeed.UPDATE)
+    changefeed.outqueue = outpipe
+    changefeed.run_forever()
+
+    assert outpipe.get()['msg'] == 'seems like we have an update here'
+    assert outpipe.qsize() == 0
+    assert mock_cursor_find_one.called_once_with(
+        {'_id': mock_changefeed_data[2]['o']},
+        {'_id': False}
+    )
+
+
+@pytest.mark.bdb
+@mock.patch('pymongo.collection.Collection.find_one')
+@mock.patch('pymongo.cursor.Cursor.alive', new_callable=mock.PropertyMock)
+@mock.patch('pymongo.cursor.Cursor.next')
+def test_changefeed_multiple_operations(mock_cursor_next, mock_cursor_alive,
+                                        mock_cursor_find_one,
+                                        mock_changefeed_data):
+    from bigchaindb.backend import get_changefeed, connect
+    from bigchaindb.backend.changefeed import ChangeFeed
+
+    conn = connect()
+    mock_cursor_alive.side_effect = [mock.DEFAULT, mock.DEFAULT,
+                                     mock.DEFAULT, mock.DEFAULT, False]
+    mock_cursor_next.side_effect = [mock.DEFAULT] + mock_changefeed_data
+    mock_cursor_find_one.return_value = mock_changefeed_data[2]['o']
+
+    outpipe = Pipe()
+    changefeed = get_changefeed(conn, 'backlog',
+                                ChangeFeed.INSERT | ChangeFeed.UPDATE)
+    changefeed.outqueue = outpipe
+    changefeed.run_forever()
+
+    assert outpipe.get()['msg'] == 'seems like we have an insert here'
+    assert outpipe.get()['msg'] == 'seems like we have an update here'
+    assert outpipe.qsize() == 0
+
+
+@pytest.mark.bdb
+@mock.patch('pymongo.cursor.Cursor.alive', new_callable=mock.PropertyMock)
+@mock.patch('pymongo.cursor.Cursor.next')
+def test_changefeed_prefeed(mock_cursor_next, mock_cursor_alive,
+                            mock_changefeed_data):
+    from bigchaindb.backend import get_changefeed, connect
+    from bigchaindb.backend.changefeed import ChangeFeed
+
+    conn = connect()
+    mock_cursor_alive.side_effect = [mock.DEFAULT, mock.DEFAULT,
+                                     mock.DEFAULT, mock.DEFAULT, False]
+    mock_cursor_next.side_effect = [mock.DEFAULT] + mock_changefeed_data
+
+    outpipe = Pipe()
+    changefeed = get_changefeed(conn, 'backlog', ChangeFeed.INSERT,
+                                prefeed=[1, 2, 3])
+    changefeed.outqueue = outpipe
+    changefeed.run_forever()
+
+    assert outpipe.qsize() == 4
+
+
+@pytest.mark.bdb
+@mock.patch('pymongo.cursor.Cursor.alive', new_callable=mock.PropertyMock)
+@mock.patch('bigchaindb.backend.mongodb.changefeed.MongoDBChangeFeed.run_changefeed')  # noqa
+def test_connection_failure(mock_run_changefeed, mock_cursor_alive):
+    from pymongo.errors import ConnectionFailure
```

Comment:
since it is not bigchaindb code, we could safely import this outside, at the top